docs/design-proposals/multi-tenancy/multi-tenancy-network.md

```diff
@@ -292,3 +303,12 @@ Components not decided yet at current phase are not included in these views.
 ### Key Scenario: tenant & DNS provisioning
 
 ![tenant & DNS provision](images/network-scenario-tenant-init.png)
+
+### Activity Flow: Pod - from creation to running
+
+![Pod-activity-diag](images/pod-activity-diagram.png)
+
+1. It is strongly recommended to annotate new pods with arktos.futurewei.com/network-readiness as hint for scheduler. It is not mandatory, though - kubelet and cni plugin will have to cope with it and gets it right eventually. This kind of initial pod annotation can be done by a custom admission control.
+2. It is the network controller's responsibility to do whatever network resource preparation for the pod; if no special interests in it, it is fine. The bottom line is the controller needs to clean the annotation arktos.futurewei.com/network-readiness=false if it is there. If pod has hint of arktos.futurewei.com/nics annoation, it should honor that.
+3. It is the network controller's responsibility to clean the annotation arktos.futurewei.com/network-readiness=false if it is there.
+4. It is the network controller's responsibility to annotate pod with proper information that the cni plugin needs in term of CNI-ARGS. Arktos system just picks up the annotation value and pass it on when calling cni plugin.
```

Comment:
I generally like this approach because it gives us a lot of decoupling. My question is that; Mizar behavior here is Arktos specific. Is there a way for us to have the same model in community Kubernetes? I'm trying to avoid having a CNI/Operator behavior for arktos that is significantly different from a regular K8s implementation. Do you have any suggestions here?

Reply:
If we community k8s able to handle following properly, then yes
1. ask scheduler temporarily put pod on hold before network controller finishes its preparation part;
2. kubelet/container runtime able to pass applicanle ani-args

for 1, either we __hack__ around with taint/toleration - which does not use network-readiness annotation, or put a custom controller which understands the annotation.
for 2, the only way I can think about is modify container runtime, as k8s has no established way to pass extra arg, though cni spec does include in its specification.